cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -681,10 +686,31 @@ private void startExecution(LoadMonitor loadMonitor,
                               boolean isTriggeredByUserRequest) throws OngoingExecutionException {
     _executionStoppedByUser.set(false);
     sanityCheckOngoingMovement();
-    _hasOngoingExecution = true;
+
+    try {
+      _flipOngoingExecutionMutex.acquire();
+      try {
+        _hasOngoingExecution = true;
+        _stopSignal.set(NO_STOP_EXECUTION);
+        try {
+          // Wait if Executor is shutting down.
+          _noOngoingExecutionSemaphore.acquire();
+          // Block Executor from shutting down until ongoing execution stopped.
+        } catch (final InterruptedException ie) {
+          _hasOngoingExecution = false;
+          throw new OngoingExecutionException("Interrupted while shutting down executor.");
```

Comment:
* `OngoingExecutionException` -> `IllegalStateException`

Note that `OngoingExecutionException` is thrown when `"...an execution was attempted to be started while there is another ongoing execution"`